proto/desmos/profiles/v1beta1/models.proto

```diff
@@ -125,24 +127,46 @@ message Proof {
 
 // ChainConfig contains the data of the chain with which the link is made.
 message ChainConfig {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
   string name = 1
       [ (gogoproto.moretags) = "yaml:\"name\"" ];
-  string bech32_addr_prefix = 2
-      [ (gogoproto.moretags) = "yaml:\"bech32_addr_prefix\"" ];
 }
 
 // ChainLink contains the data representing either an inter- or cross- chain link
 message ChainLink {
-  string address = 1 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+  
+  google.protobuf.Any address = 1 [ (gogoproto.moretags) = "yaml:\"address\"" ];
   Proof proof = 2
       [ (gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"proof\"" ];
   ChainConfig chain_config = 3 [
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"chain_config\""
   ];
-  google.protobuf.Timestamp creation_time = 4 [
+
+  google.protobuf.Timestamp creation_time = 5 [
     (gogoproto.stdtime) = true,
     (gogoproto.moretags) = "yaml:\"creation_time\"",
     (gogoproto.nullable) = false
   ];
 }
+
+ // Bech32Address represents a Bech32-encoded address
+ message Bech32Address {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  string value = 1 [ (gogoproto.moretags) = "yaml:\"value\"" ];
+  string prefix = 2 [ (gogoproto.moretags) = "yaml:\"prefix\"" ];
+}
+
+// Base58Address represents a Base58-encoded address
+message Base58Address {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
```

Comment:
Same as above, I think we should add the `implements_interface` option here as well